src/test/datascience/uiTests/webBrowserPanel.ts

```diff
@@ -186,10 +213,26 @@ export class WebBrowserPanel implements IWebviewPanel, IDisposable {
     public asWebviewUri(localResource: Uri): Uri {
         const filePath = localResource.fsPath;
         const name = path.basename(path.dirname(filePath));
-        if (name !== 'nbextensions' && this.serverUrl) {
-            // This is a CDN download, Remap to our webserver
-            const remapped = `${this.serverUrl}/source?hash=${name}`;
-            return Uri.parse(remapped);
+        if (this.serverUrl) {
+            if (name === 'nbextensions') {
+                // This is a CDN download, Remap to our webserver
+                const remapped = `${this.serverUrl}/nbextensionssource?hash=${path.basename(filePath, '.js')}`;
```

Comment:
Problem 3 - These files (in the nbextensions) folder were never getting downloaded. Hence the tests always failed when they resorted to downloaded from tmp/nbextensions folder.